### PR TITLE
GC9xxx LCD family drivers (GC9107 and GC9A01)

### PR DESCRIPTION
**Pull request by fpb**
## Description

LCD GC9XXX family of drivers similar to ST77XX. GC9107 support was added and also did a lot of initialisation code cleanup of GC9A01. Old init code for GC9A01 had a lot of flicker on my device and that has disappeared with the new (shorter) init code.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fpb** [comment]
Tested both display models and it all seems ok.

IMG_20240221_105810

**Review by drashna — APPROVED** [review]

**Comment by fpb** [comment]
Reviews are still pending...

File: drivers/painter/gc9xxx/qp_gc9107.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        0xAB, 0, 1, 0x0E,
+        GC9107_SET_FRAME_RATE, 0, 1, 0x19,
+
+        // 0xB7, 0, 1, 0x01,
+        // 0xB8, 0, 1, 0x08,
+
+
+
+        // 0xC6, 0, 1, 0x30,
+        // 0xC7, 0, 1, 0x18,
```
Comment (tzarc) [inline drivers/painter/gc9xxx/qp_gc9107.c:47]:
```suggestion
```

File: drivers/painter/gc9xxx/qp_gc9107.c [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            .display_on         = GC9XXX_CMD_DISPLAY_ON,
+            .display_off        = GC9XXX_CMD_DISPLAY_OFF,
+            .set_column_address = GC9XXX_SET_COL_ADDR,
+            .set_row_address    = GC9XXX_SET_ROW_ADDR,
+            .enable_writes      = GC9XXX_SET_MEM,
+        },
+};
+
+#ifdef QUANTUM_PAINTER_GC9107_SPI_ENABLE
+// Factory function for creating a handle to the ILI9341 device
```
Comment (tzarc) [inline drivers/painter/gc9xxx/qp_gc9107.c:99]:
```suggestion
// Factory function for creating a handle to the GC9107 device
```

**Review by tzarc — CHANGES_REQUESTED** [review]
Just a few minor tidbits.

**Review by tzarc — APPROVED** [review]

**Pull request merged**